numpy/core/setup.py

```diff
@@ -688,7 +688,10 @@ def get_mathlib_info(*args):
     config.add_installed_library('npymath',
             sources=npymath_sources + [get_mathlib_info],
             install_dir='lib',
-            build_info={'include_dirs' : []})  # empty list required for creating npy_math_internal.h
+            build_info={
+                'include_dirs' : [],
+                'extra_compiler_args' : (['/GL-'] if sys.platform == 'win32' else []),
```

Comment:
Won't this break mingw/cygwin builds?

Reply:
`sys.platform` should be `cygwin` for a cygwin build.

Reply:
For mingw builds it is win32 afaik. But I'm not 100% sure mingw build on windows even work currently.

Reply:
The `win32` platform is supposed to indicate a CRT ABI. So if mingw is setting that variable to `win32`, it's a mingw bug.

Reply:
Although technically, you're correct in that it would be better to check `c.compiler_type`. Is it worth another PR?

```
        c = distutils.ccompiler.new_compiler()
        c.customize('')
        static_exts = []
        if c.compiler_type != 'msvc':
```

Reply:
From `numpy/distutils/fcompiler/gnu.py` it seems that cygwin returns "win32" for platform, or at least did. I have now idea what it does these days. Maybe we should use the `platform` module: https://stackoverflow.com/questions/1387222/reliably-detect-windows-in-python.  Note that the platform module was new in Python 2.3, so probably after most of distutils was written.

Reply:
Or rather, we were still supporting earlier Python versions, so could not rely on the platform module being available.

Reply:
@charris #9977 resolves this.